### PR TITLE
Fix bug where a pipeline is not triggered

### LINKED ISSUE #842: Delayed Jobs
Hey guys, having some more issues with downstream commits getting triggered. I've got the following pipeline set up:

A -> B -> C -> D
and
E -> C

So everything has one input, except C, which has E and B as inputs. Everything has parallelism set to 1. Here's how the incremental flags are set up

| Pipeline | Input 1 | Incremental Flag 1 | Input 2 | Incremental Flag 2 |
| --- | --- | --- | --- | --- |
| B | A | DIFF |  |  |
| C | B | DIFF | E | DIFF |
| D | C | NONE |  |  |

Here's my issue:
1. When I first add stuff, and make my first commits to A and E, everything works fine. 
2. When I make another commit to E, C gets triggered and does what I expected. However D doesn't get triggered
3. Then I make a commit to A. B and C both get triggered as expected. D also gets triggered, but it does what i would have expected it to do after step 2. No job gets triggered for D as a result of the commit to A. 

I also tried reversing steps two and 3, and get the same issue (in reverse order). Seems like there's some sort of issue of delayed scheduling.

Also potentially relevant, I'm experience the CrashLoopBackOff issue mentioned in issue [#837](https://github.com/pachyderm/pachyderm/issues/837)

I'm running the following version of pachctl / pachd:

```
COMPONENT           VERSION                                          
pachctl             1.2.0-399ffcb287a970e65c901ff0ee3288cad173f512   
pachd               1.2.0-399ffcb287a970e65c901ff0ee3288cad173f512 
```

As an aside, what setup do you guys use to run your tests? (are there additional dependencies required, etc.) I was going to write a test to replicate this behavior, but couldn't get the test suite working.

### PR DESCRIPTION
**Pull request by derekchiang**
Fix #842

### CODE REVIEW

**Comment by jdoliner** [comment]
@derekchiang is CI failing for real reasons here?

File: src/server/pfs/db/driver.go [hunk 1]
```diff
@@ -936,17 +936,35 @@ func (d *driver) ListCommit(include []*pfs.Commit, exclude []*pfs.Commit, proven
...
+			repoToProvenance[repoInfo.Repo.Name] = append(repoToProvenance[repoInfo.Repo.Name], &persist.ProvenanceCommit{
+				Repo: commit.Repo.Name,
+				// We can't just use commit.ID directly because it might be a
+				// branch name instead of a commitID
+				ID: persist.FullClockHead(rawCommit.FullClock).ReadableCommitID(),
+			})
 		}
 	}
 
+	fmt.Printf("repoSet1: %v\n", repoSet1)
```
Comment (jdoliner) [inline src/server/pfs/db/driver.go]: Reminder to remove these `Printf`s before merging.

File: src/server/pkg/disk/monitor.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+func NewMonitor(path string) Monitor {
+	return &monitor{
+		path: path,
+	}
+}
+
+func (m *monitor) CurrentUsage() DiskUsage {
+
+}
```
Comment (jdoliner) [inline src/server/pkg/disk/monitor.go]: This seems only partially implemented was this supposed to be included?

Comment (derekchiang) [inline src/server/pkg/disk/monitor.go]: oops this is from a while ago.  old laptop stuff

**Comment by jdoliner** [comment]
aydabtu
LGTM

**Pull request merged**